framework/freedomotic-core/src/test/java/com/freedomotic/rules/StatementTest.java

```diff
@@ -0,0 +1,113 @@
+package com.freedomotic.rules;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StatementTest {
+
+  @Test
+  public void createLogicalInputValid() {
+    final Statement retval = new Statement().create(Statement.EQUALS, "retval1", Statement.AND, "100");
+    Assert.assertNotNull(retval);
+    Assert.assertEquals(Statement.EQUALS, retval.getLogical());
+    Assert.assertEquals("retval1", retval.getAttribute());
+    Assert.assertEquals(Statement.AND, retval.getOperand());
+    Assert.assertEquals("100", retval.getValue());
+  }
+
+  @Test
+  public void createLogicalInputInvalid() {
+    final Statement retval = new Statement().create(Statement.NOT, "retval2", Statement.OR, "100");
+    Assert.assertNotNull(retval);
+    Assert.assertEquals(Statement.NOT, retval.getLogical());
+    Assert.assertEquals("retval2", retval.getAttribute());
+    Assert.assertEquals(Statement.OR, retval.getOperand());
+    Assert.assertEquals("100", retval.getValue());
+  }
+
+  @Test
+  public void createStatementNull() {
+    Assert.assertNull(new Statement().create(" ", null, " ", null));
+  }
+
+  @Test
+  public void statementEqualsTrue() {
+    Assert.assertEquals(new Statement().create(null, "!", null, " "),
+                        new Statement().create(null, "!", "!", null));
+  }
+
+  @Test
+  public void statementEqualsFalse() {
+    Assert.assertNotEquals(new Statement().create(" ", null, " ", null),
+                           new Statement().create(Statement.LESS_THAN, "retval3", Statement.OR, "50"));
```

Comment:
Had a look through the uses of the `create` method, most seem to use a dot-delimited string for the `attribute` argument. Might make sense then to use a string like `"test.value"` instead of the `"retval"`-esque values currently used, since they're a bit Cover-specific.